internal/services/datafactory/data_factory_linked_service_sql_managed_instance_resource.go

```diff
@@ -0,0 +1,598 @@
+					state.KeyVaultConnectionString = flattenKeyVaultConnectionStringToConfig(val)
+				}
+				if val, ok := (*props.ConnectionString).(string); ok {
+					state.ConnectionString = val
+				}
+			}
+
+			if props.ServicePrincipalId != nil {
+				if id, ok := (*props.ServicePrincipalId).(string); ok {
+					state.ServicePrincipalID = id
+				}
+			}
+
+			if props.Tenant != nil {
+				if tenant, ok := (*props.Tenant).(string); ok {
+					state.Tenant = tenant
+				}
+			}
+
+			if v, exists := metadata.ResourceData.GetOk("service_principal_key"); exists && v.(string) != "" {
+				state.ServicePrincipalKey = v.(string)
+			}
+
+			state.KeyVaultPassword = flattenKeyVaultPasswordToConfig(props.Password)
+
+			if existing.Annotations != nil {
+				annotations := make([]string, 0)
+				for _, annotation := range *existing.Annotations {
+					if str, ok := annotation.(string); ok {
+						annotations = append(annotations, str)
+					}
+				}
+				state.Annotations = annotations
+			}
+
+			if existing.Parameters != nil {
+				parameters := make(map[string]string)
+				for key, param := range *existing.Parameters {
+					if param.DefaultValue != nil {
+						if str, ok := (*param.DefaultValue).(string); ok {
+							parameters[key] = str
+						}
+					}
+				}
+				state.Parameters = parameters
+			}
+
+			if connectVia := existing.ConnectVia; connectVia != nil {
+				state.IntegrationRuntimeName = connectVia.ReferenceName
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r LinkedServiceSqlManagedInstanceResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataFactory.LinkedServicesClient
+
+			id, err := linkedservices.ParseLinkedServiceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config LinkedServiceSqlManagedInstanceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id, linkedservices.DefaultGetOperationOptions())
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `model.Properties` was nil", *id)
+			}
+
+			payload := linkedservices.LinkedServiceResource{
+				Properties: &linkedservices.AzureSqlMILinkedService{
+					Description: pointer.To(config.Description),
+					TypeProperties: linkedservices.AzureSqlMILinkedServiceTypeProperties{
+						Password: expandKeyVaultPasswordFromConfig(config.KeyVaultPassword),
+					},
+				},
+			}
+			props := payload.Properties.(*linkedservices.AzureSqlMILinkedService)
+
+			if metadata.ResourceData.HasChanges("connection_string", "key_vault_connection_string") {
+				if config.ConnectionString != "" {
```

Comment:
is it possible to remove a connection string and use key_vault_connection_string instead?